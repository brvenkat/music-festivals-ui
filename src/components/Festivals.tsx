import * as React from 'react'
import Card from '@material-ui/core/Card';
import { CardContent } from '@material-ui/core';
import { style } from 'typestyle'

export interface Festival {
  recordName: string
  bands: Band[]
}

interface Band {
  name: string
  festivals: string[]
}

const headerStyle = style({
  textAlign: 'center',
  fontWeight: 'bold',
  fontSize: '30px'
})

const cardStyle = style({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-between',
  marginBottom: '12px',
  color: 'blue',
  border: '2px solid blue'
})

export const Festivals = (props: { festivals: Festival[]}) => {
  const allFestivals = props.festivals
  return (
    <>
    <div className={headerStyle}>Music Festival Data</div>
    {
      allFestivals.map((festival, i) => (
          <Card key={i} className={cardStyle}>
            {festival.recordName}
            <CardContent>
              {
                festival.bands.map((band, j) => (
                  <span key={j}>
                    <p>{band.name}</p>
                    {
                      band.festivals.map((f, k) => (
                        <li key={k}>{f}</li>
                      ))
                    }
                  </span>
                ))
              }
            </CardContent>
          </Card>
      ))
    }
    </>
  )
}