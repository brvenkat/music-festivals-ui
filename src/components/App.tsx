
import * as React from 'react';
import { Festivals } from './Festivals'

export function App() {
  const [festivals, setFestivals] = React.useState([])
  const [loading, setLoading] = React.useState(true);

  React.useEffect(() => {
    fetch(process.env.MUSIC_FESTIVALS_URL)
    .then(res => res.json())
    .then(response => {
      setFestivals(response.festivals)
      setLoading(false)
    })
    .catch((error) => console.log(error))
  }, [])

  if (loading) {
    return (
      <div>Loading ...</div>
    )
  }
  if (!Array.isArray(festivals) || festivals.length === 0) {
    return (
      <div> Something has gone wrong, please refresh or try after some time</div>
    )
  }
  return <Festivals festivals={festivals} />
}
