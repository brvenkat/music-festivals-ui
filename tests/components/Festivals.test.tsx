import * as React from 'react'
import { mount } from 'enzyme'
import { Festivals } from '../../src/components/Festivals'
import * as mock from '../mockData/mock'
import Card from '@material-ui/core/Card';
import { CardContent } from '@material-ui/core';

describe('Festivals ', () => {
  const mockFestival: any = mock.apiResponse
  it('should render the component correctly ', () => {
    const wrapper = mount(<Festivals festivals={mock.apiResponse} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render all components correctly ', () => {
    const wrapper = mount(<Festivals festivals={mock.apiResponse} />)
    expect(wrapper.find('div').at(0).text()).toEqual('Music Festival Data')
    expect(wrapper.find(Card).at(0).text()).toContain('record-1')
    expect(wrapper.find(CardContent).at(0).text()).toContain('name-1')
    expect(wrapper.find(CardContent).at(1).text()).toContain('festival-1')
    expect(wrapper.find(Card).at(1).text()).toContain('record-2')
    expect(wrapper.find(Card).at(1).find(CardContent).at(0).text()).toContain('name-2')
    expect(wrapper.find(Card).at(1).find(CardContent).at(0).text()).toContain('festival-1')
    expect(wrapper.find(Card).at(1).find(CardContent).at(0).text()).toContain('festival-2')
  })
})