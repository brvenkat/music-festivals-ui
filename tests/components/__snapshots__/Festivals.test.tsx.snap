// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Festivals  should render the component correctly  1`] = `ReactWrapper {}`;
